app/one_click_unsubscribe/rest.py

```diff
@@ -52,3 +60,62 @@ def get_unsubscribe_request_data(notification, email_address):
         "service_id": notification.service_id,
         "email_address": email_address,
     }
+
+
+def create_unsubscribe_request_reports_summary(service_id):
+    reports_summary = []
+    batched_unsubscribe_reports_summary = []
+    unbatched_unsubscribe_report_summary = []
+    # Check for existing unsubscribe reports and create their summaries
+    if unsubscribe_request_reports := get_unsubscribe_request_reports_dao(service_id):
+        batched_unsubscribe_reports_summary = _create_batched_unsubscribe_request_reports_summary(
+            unsubscribe_request_reports,
+        )
+        batched_unsubscribe_reports_summary = batched_unsubscribe_reports_summary
+    # Check for unsubscribe requests and create a summary for them
+    if unbatched_unsubscribe_requests := get_unbatched_unsubscribe_requests_dao(service_id):
+        if batched_unsubscribe_reports_summary:
+            earliest_timestamp = batched_unsubscribe_reports_summary[0]["latest_timestamp"]
+        else:
+            service = dao_fetch_service_by_id(service_id)
+            earliest_timestamp = service.created_at
+
+        unbatched_unsubscribe_report_summary.append(
+            _create_unbatched_unsubscribe_request_report_summary(unbatched_unsubscribe_requests, earliest_timestamp)
+        )
+
+    reports_summary = unbatched_unsubscribe_report_summary + batched_unsubscribe_reports_summary
+
+    return reports_summary
+
+
+def _create_batched_unsubscribe_request_reports_summary(unsubscribe_request_reports: list) -> list:
+    report_summaries = []
+    for report in unsubscribe_request_reports:
+        report_summary = {
+            "batch_id": str(report.id),
+            "count": report.count,
+            "earliest_timestamp": report.earliest_timestamp,
+            "latest_timestamp": report.latest_timestamp,
+            "processed_by_service_at": report.processed_by_service_at,
+            "is_a_batched_report": True,
+            "status": "Completed" if report.processed_by_service_at else "Downloaded",
```

Comment:
I think we should keep this logic in the admin app because it’s presentational, not an actual field in the database
```suggestion
```

Reply:
I understand your point. My premise was that since the summary already contained a non  database field `is_a_batched_report, however I am happy to move it to the admin app.

Reply:
Let’s say we wanted to change the wording in the future. To me it feels more natural to make that change in the admin app. So that’s why ’d suggest it lives there